services/scabbard/libscabbard/src/client/mod.rs

```diff
@@ -29,18 +29,69 @@ use super::protocol::SCABBARD_PROTOCOL_VERSION;
 
 pub use error::ScabbardClientError;
 
+/// Builder for building a ScabbardClient.
+#[derive(Default)]
+pub struct ScabbardClientBuilder {
+    url: Option<String>,
+    #[cfg(feature = "client-auth")]
+    auth: Option<String>,
+}
+
+impl ScabbardClientBuilder {
+    /// Creates a new ScabbardClientBuilder.
+    pub fn new() -> Self {
+        ScabbardClientBuilder::default()
+    }
+
+    /// Sets the `url` field of the ScabbardClientBuilder. The url will be used
+    /// as the bind endpoint for the Splinter REST API.
+    ///
+    /// # Arguments
+    ///
+    /// * `url` - The URL of the bind endpoint of the Splinter REST API.
+    pub fn with_url(mut self, url: &str) -> Self {
+        self.url = Some(url.into());
+        self
+    }
+    /// Sets the `auth` field of the ScabbardClientBuilder. The `auth` string contains a
+    /// signed json web token and will be passed to the Splinter REST API in an
+    /// Authorization header.
+    ///
+    /// # Arguments
+    ///
+    /// * `auth` - The authorization string containing a signed json web token.
```

Comment:
The auth field isn't necessarily going to be a JWT; it could be any other auth header supported by splinterd. It would be sufficient to just say that the value given here will be provided in the `Authorization` header of REST API requests.

Reply:
Fixed comment to only reference auth string and not jwt